XenonUtils/xex.cpp

```diff
@@ -192,6 +194,79 @@ Image Xex2LoadImage(const uint8_t* data, size_t dataSize)
                 destData += blocks[i].zeroSize;
             }
         }
+        else if (fileFormatInfo->compressionType == XEX_COMPRESSION_NORMAL)
+        {
+            result = std::make_unique<uint8_t[]>(imageSize);
+            auto* destData = result.get();
+
+            const Xex2CompressedBlockInfo* blocks = &((const Xex2FileNormalCompressionInfo*)(fileFormatInfo + 1))->firstBlock;
+            const uint32_t headerSize = header->headerSize.get();
+
+            const uint32_t exeLength = dataSize - headerSize;
+            const uint8_t* exeBuffer = srcData;
+
+            uint8_t* compressBuffer = NULL;
+            const uint8_t* p = NULL;
+            uint8_t* d = NULL;
+            sha1::SHA1 s;
+
+            compressBuffer = (uint8_t*)calloc(1, exeLength);
+
+            p = exeBuffer;
+            d = compressBuffer;
+
+            int resultCode = 0;
+
+            uint8_t blockCalcedDigest[0x14];
+            while (blocks->blockSize) {
+                const uint8_t* pNext = p + blocks->blockSize;
+                const auto* nextBlock = (const Xex2CompressedBlockInfo*)p;
+
+                s.reset();
+                s.processBytes(p, blocks->blockSize);
+                s.finalize(blockCalcedDigest);
+
+                if (memcmp(blockCalcedDigest, blocks->blockHash, 0x14) != 0) {
+                    resultCode = 2;
```

Comment:
What does the 2 here represent?

Reply:
My bad, meant to swap that out for the relevant return for if the decompressing fails